File: C++_Version/217Contains_Duplicate.cpp
//Given an array of integers, find if the array contains any duplicates.Your function should return true if any value appears at least twice in the array, and it should return false if every element is distinct.

#include <vector>
#include <iostream>
#include <algorithm>
#include <cstdlib>
#include <functional>
#include <string>
#include <unordered_map>
#include <map>
#include <set>
#include <unordered_set>
using namespace std;

class Solution {
public:
	bool containsDuplicate(vector<int>& nums) {
		int num_size = (int)nums.size();
		if (num_size <= 1)
			return false;

		//set<int> nums_set;
		unordered_set<int> nums_set;

		for (int i = 0; i < num_size; ++i)
		{
			if (!nums_set.insert(nums[i]).second)
				return true;
		}
		return false;
	}
};


int main()
{
	Solution sol1;
	
	vector<int> nums({ 1,2,3,4,5,6 });

	bool receive = sol1.containsDuplicate(nums);

	system("pause");

	return 0;
}

File: Java_Version/217Contains_Duplicate.java
/*Given an array of integers, find if the array contains any duplicates. Your function should return true if any value appears at least twice in the array, and it should return false if every element is distinct.*/


import sun.plugin.javascript.navig.LinkArray;

import java.util.*;
import java.lang.Math;
import java.lang.System;
import java.lang.Integer;


public class Main {

    public static void main(String[] args) throws ArithmeticException {

        ListNode root = new ListNode(1);
        root.next = new ListNode(2);

        ListNode result = new Solution().reverseList(root);

        //System.out.println(result);
    }

}


class ListNode {
    int val;
    ListNode next;

    ListNode(int x) {
        val = x;
    }
}


class TreeNode {
    int val;
    TreeNode left;
    TreeNode right;

    TreeNode(int x) {
        val = x;
    }
}


class Solution {
    public boolean containsDuplicate(int[] nums) {

        HashSet<Integer> hashSet = new HashSet<>();

        int numsLength = nums.length;

        for (int i = 0; i < numsLength; i++) {
            if (hashSet.contains(nums[i])) {
                return true;
            }

            hashSet.add(nums[i]);
        }

        return false;
    }
}